src/test/java/com/google/cloud/spanner/pgadapter/StatementTest.java

```diff
@@ -300,4 +314,87 @@ public void testPortalStatementDescribePropagatesFailure() throws Exception {
 
     intermediateStatement.describe();
   }
+
+  @Test
+  public void testBatchStatements() throws Exception {
+    String sql =
+        "INSERT INTO users (id) VALUES (1); INSERT INTO users (id) VALUES (2);INSERT INTO users (id) VALUES (3);";
+    IntermediateStatement intermediateStatement = new IntermediateStatement(sql, connection);
+
+    Assert.assertTrue(intermediateStatement.isBatchedQuery());
+    List<String> result = intermediateStatement.getStatements();
+    Assert.assertEquals(result.size(), 3);
+    Assert.assertEquals(result.get(0), "INSERT INTO users (id) VALUES (1);");
+    Assert.assertEquals(result.get(1), "INSERT INTO users (id) VALUES (2);");
+    Assert.assertEquals(result.get(2), "INSERT INTO users (id) VALUES (3);");
+  }
+
+  @Test
+  public void testAdditionalBatchStatements() throws Exception {
+    String sql =
+        "BEGIN TRANSACTION; INSERT INTO users (id) VALUES (1); INSERT INTO users (id) VALUES (2); INSERT INTO users (id) VALUES (3); COMMIT;";
```

Comment:
I assume tis will fail within JDBC because it wont accept BEGIN within PreparedStatement ?

Reply:
Yes, this is only testing the parsing of the statements.  This wouldn't work in a real batch statement.